Add package search filters for content type and selection status

wwilsman opened the pull request:
## Purpose
Adds package search filters for content type and selection status using the filters `filter[type]` and `filter[selected]` respectively.

## Approach
For the most part, copy and pasted the filter component for titles and just changed the relevant pieces to reflect the package filters. Same copy-paste method for the tests as well.

## Screenshots
![gif](https://user-images.githubusercontent.com/5005153/36399609-37e73d8a-1592-11e8-87e4-95467fbcfa30.gif)

cherewaty commented on src/components/package-search-filters.js:
```diff
@@ -0,0 +1,85 @@
+import { Accordion, FilterAccordionHeader } from '@folio/stripes-components/lib/Accordion';
+import RadioButton from '@folio/stripes-components/lib/RadioButton';
+
+const selectedFilters = [
```
Let's make all these filter labels Sentence case.

cherewaty commented:
Before we repeat this pattern again, it's probably a good time to match the `SearchAndSort` style: remove the `<hr>`s and indent the lists:

![screen shot 2018-02-19 at 4 32 05 pm](https://user-images.githubusercontent.com/230597/36399716-ccc6bea8-1592-11e8-9837-66fb135cd60f.png)

wwilsman commented:
@cherewaty good point about matching styles. I'll open another PR tomorrow to match the styles in title filters, and just rebase & re-copy the component for this PR.

wwilsman commented:
Rebased with the common search-filters component. 

While fixing the mirage packages filter, I realized that we do not need to continue filtering if the model has already been filtered out of the results. So I applied this same logic to the title filters and removed the extra boolean variable.

cherewaty commented on src/components/package-search-filters.js:
```diff
@@ -0,0 +1,45 @@
+        defaultValue: 'all',
+        options: [
+          { label: 'All', value: 'all' },
+          { label: 'Aggregated Full Text', value: 'aggregatedfulltext' },
```
Sentence case
